### PR TITLE
feat(payroll): Gusto-format provider export dropdown

### PR DESCRIPTION
**Pull request by jdelgado2002**
## Summary
- Adds a Gusto timesheet-import CSV export alongside the existing internal CSV export on the Payroll page, selectable via an "Export ▾" dropdown (shadcn `DropdownMenu`, mirrors the `Inventory.tsx` precedent).
- New `src/utils/payrollGustoExport.ts`: `GUSTO_CSV_HEADERS` (16-column Gusto timesheet-import template) + `splitEmployeeName` + `buildGustoCSV` mapper. Money fields converted from cents; `paycheck_tips` = tipsOwed, `cash_tips` = tipsPaidOut (no double-pay); `title` = position (informational); `gusto_employee_id` left blank (Gusto name-matches).
- New `src/utils/payrollExportFormats.ts`: a small format registry (`PAYROLL_EXPORT_FORMATS`) wrapping both the existing internal CSV builder and the new Gusto builder behind a shared filename/date-range helper.
- New `src/components/payroll/PayrollExportMenu.tsx`: presentational dropdown component that performs the blob download, with destructive-toast error handling (null period, builder throwing) and a `disabled` guard for empty payroll periods.
- `src/pages/Payroll.tsx` swapped its single "Export CSV" button for `<PayrollExportMenu>`.
- Security hardening carried over to shared code: broadened `escapeCsvCell` (in `src/utils/payrollCalculations.ts`, now exported) to neutralize CSV-formula-injection payloads with leading whitespace/tabs before `=`/`+`/`-`/`@`, closing a bypass found in adversarial review; `payrollGustoExport.ts` now reuses this exported helper instead of maintaining a duplicate.
- Pure client-side transform of data already on the page — no DB migrations, no edge functions, no new network surface.

## Test plan
- [x] `npm run typecheck` — clean
- [x] `npm run lint` — clean on all touched files (repo-wide pre-existing baseline unchanged)
- [x] `npm run test` — 5718 passed / 2 skipped (25 new tests across `payrollGustoExport.test.ts`, `payrollExportFormats.test.ts`, `PayrollExportMenu.test.tsx`, plus CSV-injection regression tests in `payrollCalculations.test.ts`)
- [x] `npm run test:db` — 1641/1641 pgTAP tests passed (no SQL changes; ran for completeness)
- [x] `npm run build` — succeeds
- [x] `npm run test:e2e` — 143 passed; fixed one stale selector (`employee-payroll.spec.ts`) for the new dropdown trigger; 3 pre-existing/flaky failures confirmed unrelated (reproduce identically on `main` / under worker-isolation)
- [x] Manual/UI review against CLAUDE.md Apple/Notion styling guidelines — matches `Inventory.tsx` DropdownMenu precedent
- [x] Codex adversarial review + CodeRabbit review — all findings addressed or triaged as out-of-scope/accepted risk (see design doc + progress notes)

Design doc: `docs/superpowers/specs/2026-07-07-payroll-provider-export-design.md`

## Summary by CodeRabbit

* **New Features**
  * Added a payroll export dropdown with multiple formats, including a provider-specific (Gusto) CSV export.
  * Exports now download via the payroll page with consistent, date-range-based filenames.
* **Bug Fixes**
  * Improved CSV safety by neutralizing formula-like inputs (including when preceded by whitespace).
  * More consistent handling of names, zeros, and empty fields in exported CSV content.
* **Documentation**
  * Added design and implementation planning docs for provider-specific payroll export.
* **Tests**
  * Added/updated unit and end-to-end coverage for format selection, download behavior, and CSV escaping rules.

### CODE REVIEW

File: src/utils/payrollGustoExport.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+/** Format cents as a plain dollar string with 2 decimals; zero → blank. */
+function formatGustoMoney(cents: number): string {
+  if (!cents) return '';
+  return (cents / 100).toFixed(2);
+}
+
+/** Format decimal hours to up to 2 decimals; zero → blank. */
+function formatGustoHours(hours: number): string {
+  if (!hours) return '';
+  return hours.toFixed(2);
```
[the remark this answers is not in the document]
Comment (jdelgado2002) [reply src/utils/payrollGustoExport.ts]: Fixed in a5f11509 — with a nuance the blanket "emit 0.00 for zero-valued fields" misses.

Per the Smart Import semantics you cite (zero = overwrite, blank = no-op), we now split by column ownership:

- **Columns we compute** (`regular_hours`, `overtime_hours`, `double_overtime_hours`, `paycheck_tips`, `cash_tips`) → emit an explicit `0.00` even when zero, so correcting a value down to zero **overwrites** the prior amount on re-import. This is exactly your point.
- **Columns we never populate** (`missed_break_hours`, `owners_draw`, `bonus`, `commission`, `correction_payment`, `reimbursement`, `personal_note`, and `gusto_employee_id`) → stay **blank** (no-op). Emitting `0.00` there would *clobber* values the user manages directly in Gusto (e.g. zero out someone's owner's draw or reimbursement on every re-import).

This split is robust under either interpretation of blank semantics. Tests updated (`tests/unit/payrollGustoExport.test.ts`) to pin both behaviors, and the design doc records the revised decision. Thanks — good catch on the re-import overwrite hazard.

**Comment by jdelgado2002** [comment]
Addressed the two CodeRabbit nitpicks in 8cb8ca90:

- **CSV-injection wording (design doc):** updated the spec's escaping rule to state explicitly that leading `= + - @` triggers are neutralized *including when preceded by spaces/tabs* (matching the escaper's `^\s*[=+\-@]` regex hardened earlier in this PR).
- **Negative-value test coverage:** added a `buildGustoCSV` test asserting negative computed values (e.g. `-4.50` hours, `-7.50` paycheck_tips) format correctly for payroll corrections rather than blanking, per the repo's edge-case testing guideline.
